storage/txcache/factory.go

```diff
@@ -0,0 +1,13 @@
+package txcache
+
+// CreateCache creates a (fail-safe) transactions cache
+func CreateCache(config CacheConfig) txCache {
```

Comment:
Auch. Returning an unexported interface? Why?

Reply:
Actually we often return unexported types from "constructor" functions. This is a particular instance of that practice: we return an unexported type (an interface, not a struct) in a "constructor" function (a simple factory, not an actual constructor in this case).

`txCache` was introduced as an intentionally internal interface - and it was the common denominator for `txCache`, `disabledCache` and `failsafeCache`, just to allow for a return type for `CreateCache`. It it weren't for the `disabledCache`, we would have returned the actual `TxCache` object. 

Anyway, since we plan to eagerly return errors when creating the caches - see comment above - and not have any `disabledCache` anymore, we will also drop this interface.

But in the meantime, can we keep it as unexported / internal interface?